docs/custom-flows/magic-links.mdx

```diff
@@ -1,5 +1,1071 @@
+
+<Images 
+  width={3024}
+  height={1654}
+  src="/docs/images/custom-flows/toggle-email-address.jpg"
+  alt="The 'Contact information' section of the 'Email, Phone, and Username' page in the Clerk dashboard. There is a red arrow pointing to the 'Email address' toggle, which is toggled on."
+/>
+
+If you click on the **Settings cog** icon next to **Email address**, the email address configuration screen will pop open. You can toggle on **Require** if you want to make sure that all users have an email address associated with their profile. 
+
+You can also find the **Verification methods** section on this screen. Here, you can toggle on **Email verification link** if you would like to allow your users to verify their email with a magic link. You can also toggle on **Email verification code** if you would like to allow your users to verify their email with a one-time passcode.
+
+<Callout type="info">
+  **Verification methods** are different from **Authentication factors**. **Verification methods** are used for verifying a user's identifier, such as an email address upon initial sign-up or when updating their profile. **Authentication factors** are used for authenticating a user, such as when they are signing in to your application.
+</Callout>
+
+<Images 
+  width={3024}
+  height={1654}
+  src="/docs/images/custom-flows/email-configuration-settings.jpg"
+  alt="The 'Email address' configuration settings modal. There is a red arrow pointing to the Settings cog next to 'Email address', showing where a user must click to open this modal. In this modal, there is a red arrow pointing to the 'Require' toggle, which is toggled on. There is also a red box surrounding the 'Verifcation methods' section, and within this section, there are red arrows pointing to 'Email verification link' and 'Email verification code' check boxes, both checked."
+/>
+
+## Custom flow
+
+In case one of the above integration methods doesn't cover your needs, you can make use of lower-level commands and create a completely custom magic link authentication flow.
+
+<Callout type="warning">
+  You still need to configure your instance in order to enable magic link authentication, as described at the top of this guide.
+</Callout>
+
+### Sign up using a custom flow
+
+Registration with magic links follows a set of steps that require users to enter their email address as an authentication identifier and click on a link that's delivered to them via email message.
+
+The sign-up process can be completed on the same or a different device. For example, users might enter their email address in their desktop browser, but click the sign-up magic link from their mobile phone. The user's email address will still be verified and registration will proceed.
+
+Let's see all the steps involved in more detail.
+
+- Initiate the sign-up process by collecting the user's identifier. It must be their email address.
```

Comment:
The main steps should be numbers (depending on the style guide), and then indented items could be bulleted or numbered.